src/replication.c

```diff
@@ -1105,6 +1385,9 @@ update_replica_entry(spec_t *spec, replica_t *replica, int iofd)
 
 	replica->pool_guid = ack_data->pool_guid;
 	replica->zvol_guid = ack_data->zvol_guid;
+	replica->replica_id = malloc(strlen(ack_data->replica_id)+1);
+	memset(replica->replica_id, 0, MAX_NAME_LEN);
+	strncpy(replica->replica_id, ack_data->replica_id, strlen(ack_data->replica_id));
 
 	MTX_LOCK(&spec->rq_mtx);
 	if (!is_replica_newly_connected(spec, replica)) {
```

Comment:
can we remove this one? and also `identified_replica`, and, `known_replica`

Reply:
Yes I will take it in different PR(removing replication_factor, consistency_factor from spec structure & is_replica_newly_connected )